libtextsecure/sendmessage.js

```diff
@@ -1139,26 +1150,37 @@ MessageSender.prototype = {
       expireTimer,
       profileKey,
       profile,
-      group: {
-        id: groupId,
-        type: textsecure.protobuf.GroupContext.Type.DELIVER,
-      },
+      group,
     };
 
     if (numbers.length === 0) {
-      return Promise.resolve({
+      return {
```

Comment:
return of an object here and a promise just below 
`return this.sendMessage(attrs, options);`
it's not causing any issue?

Reply:
The method is `async` so every return is implicitly wrapped in a promise.